src/nbl/video/CVulkanBuffer.h

```diff
@@ -39,6 +39,24 @@ class CVulkanBuffer : public IGPUBuffer
             return m_memory.get();
         }
 
+        void* getExternalHandle() override
+        {
+            if (m_cachedExternalHandle)
+                return m_cachedExternalHandle;
+
+            auto& ccp = getCachedCreationParams();
+
+            if (ccp.externalHandleType.value)
+            {
+                if (ccp.externalHandle)
+                    return m_cachedExternalHandle = ccp.externalHandle;
+
+                return m_cachedExternalHandle = getOriginDevice()->getExternalMemoryHandle(this);
+            }
```

Comment:
will this handle ever change?

if not, then query and cache it in the constructor and always return `m_cachedExternalHandle` (don't even make this method virtual)

Reply:
Handle will not change, even across processes it's the same.